src/test/java/org/truffleruby/integration/RubyBasicTest.java

```diff
@@ -0,0 +1,376 @@
+
+    private Value eval(String code) {
+        return context.eval("ruby", code);
+    }
+
+    @Test
+    public void testReadArray() {
+        Value value = eval("[42,2,3]");
+        assertEquals(42, value.getArrayElement(0).asInt());
+    }
+
+    @Test
+    public void testIterateArray() {
+        Value value = eval("[42,43,44]");
+        for (int i = 0; i < value.getArraySize(); i++) {
+            assertEquals(42 + i, value.getArrayElement(i).asInt());
+        }
+    }
+
+    @Test
+    public void testWriteArray() {
+        Value value = eval("[1,2,3]");
+        value.setArrayElement(0, 42);
+        assertEquals(42, value.getArrayElement(0).asInt());
+    }
+
+    @Test
+    public void testExecuteFunction() {
+        Value value = eval("proc { |a| a }");
+
+        assertEquals(42, value.execute(42).asInt());
+    }
+
+    @Test
+    public void testIsNull() {
+        assertTrue(eval("nil").isNull());
+    }
+
+    @Test
+    public void testReadMember() {
+        Value value = eval("a = Truffle::Interop.object_literal; a.foo = 42; a");
+        assertEquals(42, value.getMember("foo").asInt());
+    }
+
+    @Test
+    public void testWriteMember() {
+
+        Value value = eval("a = Truffle::Interop.object_literal; a.foo = 1; a");
+        value.putMember("foo", 42);
+        assertEquals(42, value.getMember("foo").asInt());
+    }
+
+    @Test
+    public void testReadAllMembers() {
+
+        Value value = eval("a = Truffle::Interop.object_literal(a: 42, b: 43, c: 44); a");
+        int index = 0;
+        for (String memberKey : value.getMemberKeys()) {
+            assertEquals(42 + index, value.getMember(memberKey).asInt());
+            index++;
+        }
+    }
+
+    @Ignore("GR-5515 incoming foreign writes aren't checking their caches")
+    @Test
+    public void testWriteAllMembers() {
+
+        Value value = eval("a = Truffle::Interop.object_literal(a: 1, b: 2, c: 3); a");
+        int index = 0;
+        for (String memberKey : value.getMemberKeys()) {
+            value.putMember(memberKey, 42 + index);
+            index++;
+        }
+        index = 0;
+        for (String memberKey : value.getMemberKeys()) {
+            assertEquals(42 + index, value.getMember(memberKey).asInt());
+            index++;
+        }
+    }
+
+    @Test
+    public void testInvokeFunction() {
+
+        Value value = eval("a = Truffle::Interop.object_literal; a.foo = proc { |a| a }; a");
+        assertEquals(42, value.getMember("foo").execute(42).asInt());
+    }
+
+    // ACCESS DYNAMIC HOST LANUGAGE DATASTRUCTURES IN GUEST LANGUAGE
+
+    @Test
+    public void testJavaExecutable() {
+        Value value = eval("proc { |a| a.call }");
+
+        assertEquals(42, value.execute(new ProxyExecutable() {
+            public Object execute(Value... t) {
+                return 42;
+            }
+        }).asInt());
+    }
+
+    @Test
+    public void testJavaReadMember() {
+        Value value = eval("proc { |a| a.a }");
```

Comment:
`value` is confusing here (same for other examples using a proc), can we name it `proc` or `executable` ?